framework_tvb/tvb/core/services/project_service.py

```diff
@@ -521,65 +523,123 @@ def get_datatype_details(datatype_gid):
             # We ignore exception here (it was logged above, and we want to return no details).
             return meta_atts, states, None
 
-    def _remove_project_node_files(self, project_id, gid, skip_validation=False):
+    @staticmethod
+    def _add_links_for_linked_datatypes(datatype, fk_to_project, link_to_delete, existing_dt_links):
+        # If we found a datatype that has links, we need to link those as well to the linked project
+        # so they can be also copied
+
+        linked_datatype_paths = []
+        h5_file = h5.h5_file_for_index(datatype)
+        h5.gather_all_references_by_index(h5_file, linked_datatype_paths)
+
+        for h5_path in linked_datatype_paths:
+            if h5_path in existing_dt_links:
+                continue
+
+            dt = h5.load(h5_path)
+            dt_index = h5.load_entity_by_gid(dt.gid)
+            new_link = Links(dt_index.id, fk_to_project)
```

Comment:
Could we avoid loading the `dt` and `dt_index` here, and find the gid directly from the H5 file with the `H5File.get_metadata_param` method which is static?

Reply:
But I need the id, not the gid.